salt/states/file.py

```diff
@@ -929,10 +929,10 @@ def symlink(
                     return _error(ret, ((
                                             'File exists where the backup target {0} should go'
                                         ).format(backupname)))
-                elif os.path.isfile(backupname):
-                    os.remove(backupname)
-                elif os.path.isdir(backupname):
-                    shutil.rmtree(backupname)
+                elif os.path.isfile(backupname)\
```

Comment:
From my conversation with @twangboy, I think these are necessary because of the funky way in which windows reports symlink status, but I could be wrong.